src/MIDebugEngine/Engine.Impl/Breakpoints.cs

```diff
@@ -111,7 +111,28 @@ private static MIBreakpointState StringToBreakpointState(string addr)
 
             string basename = System.IO.Path.GetFileName(documentName);     // get basename from Windows path
             basename = process.EscapePath(basename);
-            BindResult bindResults = EvalBindResult(await process.MICommandFactory.BreakInsert(basename, line, condition, ResultClass.None), pbreak);
+
+            Checksum[] checksums = null;
+#if CORECLR
+            if (process.MICommandFactory.SupportsBreakpointChecksums())
+            {
+                // TODO: This will need to be configurable in the future somehow
+                // TODO: We might want the MICommandFactory to return the algorithm we should use
+                // TODO: But HashAlgorithmID is not part of MICore becuase it uses AD7Guids
+                HashAlgorithmId[] hashAlgorithmIds = { HashAlgorithmId.SHA1Normalized };
```

Comment:
1. In this VS Case I am assuming this is NOT called on the main thread, correct? You should probably change it so that we get this stuff on the main thread along with all the other breakpoint properties. Otherwise we are adding extra thread transitions for no reason. At best that will be slow, at worst, that could hang.
2. What does the update 3 version of VSDebug do when it sees the 'SHA1Normalized' algorithm id? I am guessing we are going to need some code to either fall back to SHA1, or make it host dependent.